code/frameworks/nextjs/src/css/webpack.ts

```diff
@@ -0,0 +1,108 @@
+import { NextConfig } from 'next';
+import { getCssModuleLocalIdent } from 'next/dist/build/webpack/config/blocks/css/loaders/getCssModuleLocalIdent';
+import { cssFileResolve } from 'next/dist/build/webpack/config/blocks/css/loaders/file-resolve';
+import { Configuration as WebpackConfig } from 'webpack';
+import semver from 'semver';
+import { scopedResolve } from '../utils';
+
+export const configureCss = (baseConfig: WebpackConfig, nextConfig: NextConfig): void => {
+  const rules = baseConfig.module?.rules;
+  rules?.forEach((rule, i) => {
+    if (typeof rule !== 'string' && rule.test instanceof RegExp && rule.test.test('test.css')) {
+      rules[i] = {
+        test: /\.css$/,
+        use: [
+          'style-loader',
+          {
+            loader: 'css-loader',
+            options: {
+              importLoaders: 1,
+              ...getImportAndUrlCssLoaderOptions(nextConfig),
+              modules: {
+                auto: true,
+                getLocalIdent: getCssModuleLocalIdent,
+              },
+            },
+          },
+          'postcss-loader',
+        ],
+      };
+    }
+  });
+  rules?.push({
+    test: /\.(scss|sass)$/,
+    use: [
+      'style-loader',
+      {
+        loader: 'css-loader',
+        options: {
+          importLoaders: 3,
+          ...getImportAndUrlCssLoaderOptions(nextConfig),
+          modules: { auto: true, getLocalIdent: getCssModuleLocalIdent },
+        },
+      },
+      'postcss-loader',
+      'resolve-url-loader',
+      {
+        loader: 'sass-loader',
+        options: {
+          sourceMap: true,
+          sassOptions: nextConfig.sassOptions,
+          additionalData:
+            nextConfig.sassOptions?.prependData || nextConfig.sassOptions?.additionalData,
+        },
+      },
+    ],
+  });
+};
+
+/**
+ * webpack v4-v6 api
+ * https://webpack.js.org/loaders/css-loader/#url
+ * https://webpack.js.org/loaders/css-loader/#import
+ *
+ * webpack v3 api
+ * https://webpack-3.cdn.bcebos.com/loaders/css-loader/#url
+ * https://webpack-3.cdn.bcebos.com/loaders/css-loader/#import
+ */
+const getImportAndUrlCssLoaderOptions = (nextConfig: NextConfig) =>
```

Comment:
its worth checking if with the upgrade of storybook to v7 we can get rid of css loader v4 support.

Reply:
Maybe.... it wasn't that some people were using webpack 4. My addon never supported webpack 4. It was just that some people because of npm versiom resolution were using various css loader major versions. I believe part of the problem was from some storybook internal dependencies but I can't remember.

Reply:
https://github.com/RyanClementsHax/storybook-addon-next/issues/22 here is one of the related issues